auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -384,11 +385,15 @@ public static int getDefaultTheme() {
             }
 
             String type = ProviderUtils.providerIdToAccountType(userInfo.getProviderId());
+            if (type != null) {
+                credentials.add(CredentialsUtils.buildCredentialOrThrow(user, null, type));
+            }
+        }
 
-            credentials.add(new Credential.Builder(
-                    user.getEmail() == null ? user.getPhoneNumber() : user.getEmail())
-                    .setAccountType(type)
-                    .build());
+        // Also add an email credential to ensure password accounts get deleted.
```

Comment:
This confuses me a bit ... feels wrong that the `getCredentialsFromFirebaseUser` method will always return a password credential with a made-up password.

Seems like something that could surprise us down the road, and it makes the method contract kind of a lie.

Reply:
Huge thank you on catching that one! Refactor mistake, I've fixed it.